src/main/java/duke/system/Storage.java

```
@@ -0,0 +1,130 @@
+                    item.isDone();
+                }
+                tasks.add(item);
+            } else if (line.contains("\uD835\uDD3B")) { //Deadline
+                String taskLine = line.substring((line.indexOf("\uD835\uDD3B") + 5));
+                String[] descriptionAndBy = taskLine.split(" \\(by: ");
+                descriptionAndBy[1] = descriptionAndBy[1].substring(0, descriptionAndBy[1].length()-1);
+                LocalDate date = findDate(taskLine);
+                Deadline item = new Deadline(descriptionAndBy[0], descriptionAndBy[1], date);
+                if (date != null) {
+                    item.hasDate();
+                }
+                if (line.contains("\u2713")) {
+                    item.isDone();
+                }
+                tasks.add(item);
+            }
+        }
+    }
+
+    /**
+     * Saves the existing list of tasks into the duke.txt file
+     *
+     * @param tasks list of tasks
+     * @throws IOException when there is no input
+     */
+    public static void writeToFile(ArrayList<Task> tasks) throws IOException {
+        FileWriter fileWriter = new FileWriter("duke.txt");
+        fileWriter.write("Here is the collated list of your tasks:" + System.lineSeparator());
+        int i = 0;
+        for (Task task : tasks) {
+            fileWriter.append("(").append(String.valueOf(i + 1)).append(") ").append(String.valueOf(task)).
+                    append(System.lineSeparator());
+            i++;
+        }
+        fileWriter.close();
+    }
+
+    public static LocalDate findDate(String input) {
+        String[] words = input.split(" ");
+        for (int i = 0; i < words.length; i++) {
+            try {
+                return LocalDate.parse(String.format("%s %s %s", words[i], words[i + 1], words[i + 2]),
+                        DateTimeFormatter.ofPattern("MMM dd yyyy)"));
+            } catch (DateTimeParseException | ArrayIndexOutOfBoundsException ignored) {
```

Avoid empty catch blocks. It is always better to provide user with some information on why the exception was raised here.